### PR TITLE
iotools: Init at 1.6

### PR DESCRIPTION
**Pull request by felixsinger**
###### Motivation for this change
Provides a set of simple command line tools which allow access to hardware device registers. Supported register interfaces include PCI, IO, memory mapped IO, SMBus, CPUID, and MSR. Also included are some utilities which allow for simple arithmetic, logical, and other operations.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3043,6 +3043,12 @@
     githubId = 9959940;
     name = "Andreas Fehn";
   };
+  felixsinger = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:3046]: Please put this into a separate commit with the message ``maintainers: add felixsinger``.

File: pkgs/tools/misc/iotools/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    sha256 = "0vymnah44d5bzsjhfmxkcrlrikkp0db22k7a1s8bknz7glk9fldn";
+  };
+
+  nativeBuildInputs = [ gnumake gcc10 ];
+
+  makeFlags = [ "DEBUG=0" "STATIC=0" ];
+
+  installPhase = ''
+    mkdir -p $out/usr/bin
+    cp iotools $out/usr/bin/
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/iotools/default.nix:22]:
```suggestion
    install -Dm755 iotools $out/usr/bin/
```

File: pkgs/tools/misc/iotools/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  '';
+
+  meta = with lib; {
+    description = ''
+      Provides a set of simple command line tools which allow access to
+      hardware device registers. Supported register interfaces include PCI,
+      IO, memory mapped IO, SMBus, CPUID, and MSR. Also included are some
+      utilities which allow for simple arithmetic, logical, and other
+      operations.
+    '';
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/iotools/default.nix:32]:
```suggestion
    description = "Set of simple command line tools which allow access to
      hardware device registers";
    longDescription = ''
      Provides a set of simple command line tools which allow access to
      hardware device registers. Supported register interfaces include PCI,
      IO, memory mapped IO, SMBus, CPUID, and MSR. Also included are some
      utilities which allow for simple arithmetic, logical, and other
      operations.
    '';
```

File: pkgs/tools/misc/iotools/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    homepage = "https://github.com/adurbin/iotools";
+    license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/iotools/default.nix:34]: gpl2Only or gpl2Plus?

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (felixsinger) [reply maintainers/maintainer-list.nix:3046]: It is a seperate commit, see https://github.com/NixOS/nixpkgs/pull/111632/commits/a29eeb386efa82e51bf7af37ac4446932fc68083. Github just displays it in a confusing way.

File: pkgs/tools/misc/iotools/default.nix (hunk 4 above)
Comment (felixsinger) [reply pkgs/tools/misc/iotools/default.nix:34]: COPYING says GPLv2, so I think gpl2Only

File: pkgs/tools/misc/iotools/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+  src = fetchFromGitHub {
+    owner = "adurbin";
+    repo = pname;
+    rev = "18949fdc4dedb1da3f51ee83a582b112fb9f2c71";
+    sha256 = "0vymnah44d5bzsjhfmxkcrlrikkp0db22k7a1s8bknz7glk9fldn";
+  };
+
+  nativeBuildInputs = [ gnumake gcc10 ];
+
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/iotools/default.nix:17]:
```suggestion

```

Sorry that I did not notice this earlier but they should be included in stdenv already.

Comment (felixsinger) [reply pkgs/tools/misc/iotools/default.nix]: Ah okay. I removed it.

File: pkgs/tools/misc/iotools/default.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "iotools";
+  version = "1.6";
+
+  src = fetchFromGitHub {
+    owner = "adurbin";
+    repo = pname;
+    rev = "18949fdc4dedb1da3f51ee83a582b112fb9f2c71";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/iotools/default.nix:12]:
```suggestion
    rev = "v{version}";
```

File: pkgs/tools/misc/iotools/default.nix [hunk 7]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, lib, fetchFromGitHub
+, gnumake, gcc10
+}:
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/iotools/default.nix:3]:
```suggestion
{ stdenv, lib, fetchFromGitHub }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Sorry that I did not notice this earlier but after that it should be good to go.

**Review by felixsinger — COMMENTED** [review]
a

File: pkgs/tools/misc/iotools/default.nix (hunk 6 above)
Comment (felixsinger) [reply pkgs/tools/misc/iotools/default.nix:10]: If it doesn't hurt too much, I would like to use the git revision since iotools got many improvements since v1.6.

Comment (SuperSandro2000) [reply pkgs/tools/misc/iotools/default.nix:10]: Then the version should be changed to unstable-XXXX-XX-XX where XXXX-XX-XX is the date of the commit.

Comment (felixsinger) [reply pkgs/tools/misc/iotools/default.nix:10]: Can I also use the first chars of the git revision id?

Comment (SuperSandro2000) [reply pkgs/tools/misc/iotools/default.nix:10]: I am not sure. This is usually only done if upstream does it.

Comment (sikmir) [reply pkgs/tools/misc/iotools/default.nix:10]: > Can I also use the first chars of the git revision id?

https://nixos.org/manual/nixpkgs/stable/#sec-package-naming
> If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23".

File: pkgs/tools/misc/iotools/default.nix [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    owner = "adurbin";
+    repo = pname;
+    rev = "18949fdc4dedb1da3f51ee83a582b112fb9f2c71";
+    sha256 = "0vymnah44d5bzsjhfmxkcrlrikkp0db22k7a1s8bknz7glk9fldn";
+  };
+
+  makeFlags = [ "DEBUG=0" "STATIC=0" ];
+
+  installPhase = ''
+    install -Dm755 iotools $out/usr/bin/iotools
```
Comment (sikmir) [inline pkgs/tools/misc/iotools/default.nix:17]:
```suggestion
    install -Dm755 iotools -t $out/bin
```

**Review by SuperSandro2000 — APPROVED** [review]
